src/tests/ftest/object/same_key_different_value.py

```diff
@@ -176,63 +159,58 @@ def test_array_to_single_value(self):
         for i in range(3):
             try:
                 # create an object and write array value data into it
-                obj = self.container.write_an_array_value(array_value_data,
-                                                          dkey, akey,
-                                                          obj_cls=1)
+                obj = self.container.container.write_an_array_value(
+                    array_value_data, dkey, akey, obj_cls=1)
                 # read the data back and make sure its correct
                 length = len(array_value_data[0])
-                read_back_data = self.container.read_an_array(
+                read_back_data = self.container.container.read_an_array(
                     len(array_value_data), length + 1, dkey, akey, obj)
 
                 for j in range(3):
-                    if (array_value_data[j][0:length-1] !=
-                            read_back_data[j][0:length-1]):
+                    if array_value_data[j][0:length - 1] != read_back_data[j][0:length - 1]:
                         self.log.info("Written Data: %s", array_value_data[j])
                         self.log.info("Read Data:    %s", read_back_data[j])
-                        self.fail("Data mismatch\n")
+                        self.fail("Data mismatch")
                 # test case 1
                 if i == 0:
                     try:
                         # write single value data to same keys, expected to fail
-                        self.container.write_an_obj(single_value_data,
-                                                    len(single_value_data)+1,
-                                                    dkey, akey, obj, obj_cls=1)
+                        self.container.container.write_an_obj(
+                            single_value_data, len(single_value_data) + 1,
+                            dkey, akey, obj, obj_cls=1)
                         # above line is expected to return an error,
                         # if not fail the test
                         self.fail("Single value write to existing array value"
                                   + " key should have failed\n")
 
                     # should fail with -1001 ERR
-                    except DaosApiError as excp:
-                        if "-1001" not in str(excp):
-                            self.log.info(excp)
-                            self.fail("Should have failed with -1001 error"
-                                      + " message, but it did not\n")
+                    except DaosApiError as error:
+                        if "-1001" not in str(error):
+                            self.log.error(error)
+                            self.fail("Expected write to fail with -1001")
```

Comment:
Given that it appears to display locals in the stacktrace I do wonder if a simpler 'assert "-1001" in str(error)' would work here.

Reply:
We could do this if I need to push again:
```
self.assertIn('-1001', str(error), Expected write to fail with -1001)
```